lib/publiccloud/instance.pm

```diff
@@ -430,7 +430,8 @@ sub wait_for_ssh {
     # result display
     $sysout .= "\nTimeout $args{timeout} sec. expired" if ($duration >= $args{timeout});
     $instance_msg = "Check" . ($args{systemup_check} ? " SYSTEM " : " SSH ") . ($args{wait_stop} ? "DOWN" : "UP") .
-      ", $instance_msg, Duration: $duration sec.\nResult: $sshout . $sysout";
+      ", $instance_msg, Duration: $duration sec.\nResult: $sshout";
+    $instance_msg .= $sysout if defined($sysout);
```

Comment:
Ok, but kind question: in old version when sysout was undefined it simply printed *nothing* unconditionally: was there any issue with it?

Reply:
Also present `sysout` below in `croak`

Reply:
```suggestion
      ", $instance_msg, Duration: $duration sec.\nResult: $sshout" . $sysout;
```
this way, when sysout undefined, nothings more added.

Reply:
yes perl is throwing exception about attempt to use undefined variable which confusing reviewer when you try to find real problems in log of the test

Reply:
ok, a bit strange because sysout is declared in L#364: may be the issue is the double-quote enveloping the variable, but ok with that change.

Reply:
> ok, a bit strange because sysout is declared in L#364:

you are totally right in line 364 variable is **declared** and when variable is declared but not **defined** it value is **undefined** which is actual root cause of problem which I am trying to fix with my PR